pkg/merger/merger_test.go

```diff
@@ -408,6 +408,71 @@ func Test_MergeAndUpdate(t *testing.T) {
 		})
 	}
 }
+func Test_RecordLastModified(t *testing.T) {
+	cases := []struct {
+		name   string
+		attrs  *storage.ReaderObjectAttrs
+		mets   *Metrics
+		source string
+	}{
+		{
+			name:   "nil attrs; succeeds",
+			attrs:  nil,
+			mets:   &Metrics{},
+			source: "fake-source",
+		},
+		{
+			name:   "nil mets; succeeds",
+			attrs:  &storage.ReaderObjectAttrs{},
+			mets:   nil,
+			source: "fake-source",
+		},
+		{
+			name:   "nil; succeeds",
+			attrs:  nil,
+			mets:   nil,
+			source: "",
+		},
+	}
+
+	for _, tc := range cases {
+		t.Run(tc.name, func(t *testing.T) {
+			recordLastModified(tc.attrs, tc.mets, tc.source)
+		})
+	}
+}
+
+func Test_RecordLastModified_RecordMetrics(t *testing.T) {
+	var fakeMetric FakeInt64
+
+	attrs := &storage.ReaderObjectAttrs{
+		LastModified: time.Now(),
+	}
+	mets := &Metrics{
+		LastModified: &fakeMetric,
+	}
+	source := ""
+
+	recordLastModified(attrs, mets, source)
+	if fakeMetric.last != 0 {
+		t.Errorf("recordLastModified() fail; got %d, want %d", fakeMetric.last, 0)
+	}
+}
+
+type FakeInt64 struct {
+	read bool // Fake implementation not concurrent-safe
+	last int64
+}
+
+func (f *FakeInt64) Name() string {
+	return "FakeInt64"
+}
+
+func (f *FakeInt64) Set(n int64, _ ...string) {
+	if f.read {
```

Comment:
I don't think you ever set f.read to true in these tests, so I'm not sure if this fake works. The test is just checking for a [zero-value](https://go.dev/tour/basics/12) so it's hard to tell.

I might expect a boolean to see if "Set" was called. Something like
```
f.set = true
f.last = n
```
... then your tests can just check if the fake metric was set and what it was set to.